includes/Core/Authentication/Authentication.php

```diff
@@ -466,6 +476,11 @@ private function inline_js_setup_data( $data ) {
 		$data['requiredScopes']     = $auth_client->get_required_scopes();
 		$data['grantedScopes']      = ! empty( $access_token ) ? $auth_client->get_granted_scopes() : array();
 		$data['needReauthenticate'] = $data['isAuthenticated'] && $this->need_reauthenticate();
+		$data['errorCode']          = $this->user_options->get( OAuth_Client::OPTION_ERROR_CODE );
+
+		if ( $data['errorCode'] && $auth_client->using_proxy() ) {
```

Comment:
I see what you mean. I wonder if the `get_error_message()` should simply be removed here and only set the error code? Particularly because in the `default` case, it can potentially return html to [Re-authenticate with Google](#) which would be escaped by React. 

How about we remove the html link from the error message in the default case, and move the link into the `oauth_error` notice instead under the same conditions? I believe the link is only needed in the notice displayed on non-Site Kit screens anyways so it shouldn't need to be part of the message. If we keep the contract of the return as message-only, without html, then we can continue to use this without a problem and set `errorMessage` as you said.

Also, it seems a bit strange that the user's access code is deleted in some cases within this method; definitely an unexpected side effect.

Let me know how you'd like to proceed here.

Reply:
Sounds good. As discussed, let's move the entire logic related to whether there is an access code and outputting the related link to the notice rendering itself. When rendering that notice it makes much more sense to delete the access code from the DB (similar to how the error code is deleted from the DB there).

Reply:
Take a look now. I wonder if the `OAuth_Client::OPTION_ERROR_CODE` should be deleted in `authorize_user` rather than after the first time it's rendered? That, and maybe it should be dismissible (similar to the beta migration) where the ajax callback simply deletes the error code?